### PR TITLE
Add capability to deal with stations with no identifier from Wyoming

### LINKED ISSUE #238: Accessing Wyoming Soundings no ICAO Station ID
So I was attempting to access data from March of 1976 from the University of Wyoming Sounding archive and when going back that far, not every station is associated with a three-letter ICAO ID on the site. A number of the stations are listed by number. Okay not terrible, but not great. I put the number in instead of the station ID and it finds the right data file, but due to the lack of a Station ID, the parsing info from below "Station information and sounding indices" it doesn't work correctly (http://weather.uwyo.edu/cgi-bin/sounding?region=naconf&TYPE=TEXT%3ALIST&YEAR=1976&MONTH=03&FROM=0400&TO=0400&STNM=72349) compared to (http://weather.uwyo.edu/cgi-bin/sounding?region=naconf&TYPE=TEXT%3ALIST&YEAR=1976&MONTH=03&FROM=0400&TO=0400&STNM=72456) the different is in the lack of the "Station Identifier:" line in the 72349 station file.

There might be a relatively easy way to make the parsing of that first line more amenable to working with a set of data that doesn't contain a Station Identifier line. The following code will give the error that results from the previous description.

```python
from datetime import datetime

from siphon.simplewebservice.wyoming import WyomingUpperAir

date = datetime(1976, 3, 4, 0)

data = WyomingUpperAir.request_data(date, '72349')
```

### PR DESCRIPTION
**Pull request by jrleeman**
Closes #238 

Deals with missing lines from the wyoming archive when the station has a number, but no identifier.

### CODE REVIEW

File: siphon/simplewebservice/wyoming.py [hunk 1]
```diff
@@ -81,6 +81,12 @@ def _get_data(self, time, site_id):
         # Parse metadata
         meta_data = soup.find_all('pre')[1].contents[0]
         lines = meta_data.splitlines()
+
+        # If there are only 26 lines, the station doesn't have a name identified
+        # and we need to insert a record showing this for parsing to proceed.
+        if len(lines) == 26:
```
Comment (dopplershift) [inline siphon/simplewebservice/wyoming.py]: You couldn't instead check:
```python
lines[1].startswith('Station identifier')
```
?

Comment (jrleeman) [reply siphon/simplewebservice/wyoming.py]: Could sure

**Comment by jrleeman** [comment]
Happy to change it, though I don't really see it being more or less brittle as any other weirdness in the footer will still break things. I'll revise now.

File: siphon/simplewebservice/wyoming.py (hunk 1 above)
Comment (dopplershift) [reply siphon/simplewebservice/wyoming.py]: Just seems like it could change from 26 lines for a variety of reasons.

Comment (jrleeman) [reply siphon/simplewebservice/wyoming.py]: Sure, but we'll break on all of  them 😭 
Push coming.

**Comment by jrleeman** [comment]
Ok, not quite as easy as your `startswith` because we have a lot of leading whitespace and the field is totally missing, not just unpopulated. Anywho, this should do the trick.

**Comment by dopplershift** [comment]
Right, if it's missing then (now accounting for whitespace) `lines[1].strip().startswith('Station identifier:')` should return `False`, no? Your new way is fine, I just wasn't sure if there was something else I'm missing.

**Comment by jrleeman** [comment]
The whitespace... It would have to have 28 leading whitespaces on the string. Ugly.

**Comment by dopplershift** [comment]
That’s fine. But my original suggestion was to look for “Station Identifier” to be missing from `lines[1]`, yours is to look only for “Station number” there. I just was confused as to why my approach, which is checking directly for the case that you’re solving, wouldn’t work.

Either way is fine, and I don’t want to bike she’d this to death. Just trying to see if I’m missing something.

**Comment by jrleeman** [comment]
I just think this is a bit clearer and they both accomplish the same task with the same complexity as far as I can tell. Tomato vs Tomato?

**Pull request merged**